### ISSUE TITLE
[BUG] If project namespace is terminating, creating a CRTB produces an error in the logs

### ISSUE THREAD

**Issue by JonCrowther** [rancher/rancher#42904]
**Rancher Server Setup**
- Rancher version: 2.8.0
- Installation option (Docker install/Helm Chart): local 

**Information about the Cluster**
- Kubernetes version: v1.27.5 +rke2r1
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? Digital Ocean

**User Information**
- What is the role of the user logged in? Admin

**Describe the bug**
If there is a project namespace in the terminating state, creating a CRTB will attempt to create a rolebinding in the project namespace but it can't because it's terminating. It continuously requeues this operation which produces errors in the logs.

If you are able to successfully delete the namespace, the operation continues attempting to create a rolebinding in a non-existent namespace, which continues adding errors to the logs.

**To Reproduce**
1. Create a downstream cluster
2. Create a project in the downstream cluster
3. Add a finalizer to the project namespace
4. Delete the project namespace (it gets stuck in terminating)
5. Create a new user
6. Add that user as a cluster-owner to the downstream cluster
7. (For second error message) Remove the finalizer from the project namespace 

**Result**

The initial error in the logs:
```
[ERROR] error syncing 'c-m-qkdw676w/crtb-9w8b9': handler mgmt-auth-crtb-controller: couldn't create role cluster-owner: roles.rbac.authorization.k8s.io "cluster-owner" is forbidden: unable to create new content in namespace p-h7fs5 because it is being terminated, requeuing
```

After removing the finalizer:
```
[ERROR] error syncing 'c-m-qkdw676w/crtb-9w8b9': handler mgmt-auth-crtb-controller: couldn't create role cluster-owner: namespaces "p-h7fs5" not found, requeuing
```

**Expected Result**
If in the process of creating a CRTB, a project namespace is found to be terminating, it shouldn't requeue the operation. Instead it should assume that that project namespace is invalid and continue on with the process.

**Comment by JonCrowther** [comment]
# Validation

## Root Cause

When a namespace is terminating, it deletes all the resources inside it before it gets deleted itself. During this process, it is not possible to create new resources in the namespace. When we attempt to create management plane CRTB and PRTBs we only check that the namespace exists, and if it does, we attempt to create the CRTB and PRTB. This causes errors to show up in the logs while the namespace is terminating.

As a secondary effect, some of the integration tests will fail if the system is slow and API calls happen while a namespace is terminating and the test see errors. This is rare, but happens when the system experiences high load.

## What was fixed, or what change has occurred

A check was added to the management CRTB and PRTBs to only attempt to create them when the namespace is active and not terminating.

## Areas or cases that should be tested

- Creating and deleting projects and namespaces.
- Adding and removing cluster-owners 

## What areas could experience regressions

Unlikely, but any existing tests for CRTB/PRTBs, Namespaces, Projects and Cluster-Owners would be a good check.

## Are the repro steps accurate/minimal?

For testing, I would add the following finalizer for the project namespace:

```
...
finalizers:
  - kubernetes
...
```

**Comment by Priyashetty17** [comment]
<a name="top"></a>

## QA TEST PLAN

In addition to "**existing RBAC regression tests**", following tests will be executed to validate the fix:
### Test Cases

| \#  | Priority | Description & Link                                                      | PASS/FAIL |
| --- | -------- | ----------------------------------------------------------------------- | --------- |
| 1   | P0       | [User Addition to Cluster with Terminating Project Namespace](#test-1)  |           |
| 2   | P0       | [User Addition to Project with Terminating Project Namespace](#test-2)  |           |


<details>
    <summary>🚨 2 test cases... CLICK TO EXPAND! ⬅️</summary>

<br>

# 1 User Addition to Cluster with Terminating Project Namespace  

<a name="test-1"></a>

**:small_red_triangle: [back to top](#top)**

<details>
    <summary>Test 1 details... Click to expand</summary>

**Test Steps:**
1. Create a downstream cluster in Rancher.
2. Create a standard user.
3. Create a project "testProject1" in the downstream cluster.
4. Simulate a project stuck in terminating state by adding a finalizer to the project namespace "controller.cattle.io/dummy".
5. Delete the project "testProject1". 
6. Add the user as a cluster-owner to the downstream cluster.
7. Force delete the project stuck in terminating state by removing the finalizer "controller.cattle.io/dummy" from the project namespace 
8. Verify that the project is deleted.
9. Verify that there are no errors logged indicating role binding creation when a project namespace is in terminating state

</details>

<hr>

# 2 User Addition to Project with Terminating Project Namespace  

<a name="test-2"></a>

**:small_red_triangle: [back to top](#top)**

<details>
    <summary>Test 2 details... Click to expand</summary>

**Test Steps:**
1. Create a downstream cluster in Rancher.
2. Create a standard user.
3. Create a project "testProject1" in the downstream cluster.
4. Simulate a project stuck in terminating state by adding a finalizer to the project namespace "controller.cattle.io/dummy".
5. Delete the project "testProject1". 
6. Attempt to add the user as a project-owner to the project "testProject1". Verify that this fails with 403 Forbidden error.
7. Force delete the project stuck in terminating state by removing the finalizer "controller.cattle.io/dummy" from the project namespace 
8. Verify that the project is deleted.

</details>

<hr>

</details>

**Comment by Priyashetty17** [comment]
@joesims22  I have already automated Tests #1 and #2 in the test plan ^^ and is currently in review . You may execute the upgrade test when this issue is ready to test.

**Comment by Priyashetty17** [comment]
@JonCrowther Re-opening this issue based on our discussion. Original error related to "`couldn't create role cluster-owner`" has been resolved in the latest v2.9-head. However, the Rancher log is being spammed with the following error after adding a user as cluster owner to the downstream cluster with a project in terminating state:
```
2024/02/08 19:00:53 [ERROR] error syncing 'c-m-xvvlvzvn/crtb-5fxk8': handler mgmt-auth-crtb-controller: couldn't get namespace p-c6tht: namespaces "p-c6tht" not found, requeuing
Thu, Feb 8 2024 2:01:23 pm2024/02/08 19:01:23 [ERROR] error syncing 'c-m-xvvlvzvn/crtb-5fxk8': handler mgmt-auth-crtb-controller: couldn't get namespace p-c6tht: namespaces "p-c6tht" not found, requeuing
Thu, Feb 8 2024 2:01:53 pm2024/02/08 19:01:53 [ERROR] error syncing 'c-m-xvvlvzvn/crtb-5fxk8': handler mgmt-auth-crtb-controller: couldn't get namespace p-c6tht: namespaces "p-c6tht" not found, requeuing
Thu, Feb 8 2024 2:02:23 pm2024/02/08 19:02:23 [ERROR] error syncing 'c-m-xvvlvzvn/crtb-5fxk8': handler mgmt-auth-crtb-controller: couldn't get namespace p-c6tht: namespaces "p-c6tht" not found, requeuing
Thu, Feb 8 2024 2:04:23 pm2024/02/08 19:04:23 [ERROR] error syncing 'c-m-xvvlvzvn/crtb-5fxk8': handler mgmt-auth-crtb-controller: couldn't get namespace p-c6tht: namespaces "p-c6tht" not found, requeuing
```

**Note**: The error stops getting logged when the dummy finalizer is removed from the project and the project gets deleted.

**Comment by joesims22** [comment]
Validated on v2.9-head `c041edd`. All scenarios identified in the test plan have been executed and passed ✅
